backend/substrapp/ledger_utils.py

```diff
@@ -106,8 +106,13 @@ def _raise_for_status(response):
     """Parse ledger response and raise exceptions in case of errors."""
     if not response or 'error' not in response:
         return
+
+    if 'cannot change status' in response['error']:
+        raise LedgerStatusError.from_response_dict(response)
```

Comment:
@Kelvin-M quick sanity check: did you test that this `raise` is indeed executed?

Reply:
Yes :)

``` python
response = _call_ledger('invoke', fcn, args)
[response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb5599
2cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\
003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"erro
r\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb5599
2cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\
003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"erro
r\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
, response {
  status: 500
  message: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
  payload: "{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}"
}
payload: "\n \326O\335\227\227\003W\000\273`\031$\327!\240^\217<5\312\3072\341\227\335\025\024n\252\331\222\313\022\330\004\n\375\001\022\024\n\004lscc\022\014\n\n\n\004mycc\022\002\010\014\022\344\001\n\004mycc\022\333\001\nG\n@2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331\022\003\010\325\n\nG\n@8d82c648c61481f5e2b4efec27e548e3c81539372ad3c4513fd49f6416920b8f\022\003\010\232\031\nG\n@fa1813e61c13e5267e26874485f419e12f3bd5ceedd943a846b5a6527a77c9cc\022\003\010\230\035\032\315\002\010\364\003\022\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\032\242\001{\"error\":\"update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing\",\"status\":500}\"\006\022\004mycc"
]
---------------------------------------------------------------------------
LedgerStatusError                         Traceback (most recent call last)
<ipython-input-9-eaddb07d6156> in <module>
----> 1 response = _call_ledger('invoke', fcn, args)

<ipython-input-8-2f7be6703c84> in _call_ledger(call_type, fcn, args, kwargs)
     76
     77         # Raise errors if status is not ok
---> 78         _raise_for_status(response)
     79
     80         return response

<ipython-input-7-d92d7dbbabbc> in _raise_for_status(response)
      5
      6     if 'cannot change status' in response['error']:
----> 7         raise LedgerStatusError.from_response_dict(response)
      8
      9     status_code = response['status']

LedgerStatusError: update aggregatetuple 2afcb55992cd514645c6badadf6e3cd50f0fa0ba89cd67263918fc230d2ef331 failed: cannot change status from waiting to doing
```

Reply:
:propre: